### PR TITLE
[GEOS-6172] Unable to seed layer with style assigned to workspace

### PR DESCRIPTION
**Pull request by aaime**

### CODE REVIEW

File: src/gwc/src/main/java/org/geoserver/gwc/layer/CatalogStyleChangeListener.java [hunk 1]
```diff
@@ -78,17 +78,24 @@ public void handleModifyEvent(CatalogModifyEvent event) throws CatalogException
             return;
         }
         final List<String> propertyNames = event.getPropertyNames();
-        if (!propertyNames.contains("name")) {
+        if (!propertyNames.contains("name") && !propertyNames.contains("workspace")) {
```
Comment (smithkm) [inline src/gwc/src/main/java/org/geoserver/gwc/layer/CatalogStyleChangeListener.java]: Does this catch both moving to a new workspace and the workspace's name being changed?

Comment (smithkm) [inline src/gwc/src/main/java/org/geoserver/gwc/layer/CatalogStyleChangeListener.java]: Does this catch both moving to a new workspace and the name of the workspace changing?

**Comment by smithkm** [comment]
Any progress?  If you're short on time for this I could tackle the workspace rename case.

**Comment by aaime** [comment]
Ah, this is not a funded development, so you won't see me work on it during the working week. I'll resume  working on it tomorrow, but if you have spare time and want to add the workspace renaming support be my guest :)

**Comment by smithkm** [comment]
Cool, we have some internal use for both it and the layer group patch.  My issue is fitting it in between other priorities.  I'll see what I can do, and failing that I can at least offer fairly prompt review.

**Comment by aaime** [comment]
Pull request updated to handle workspace renames. Review appreciated :-)

**Comment by smithkm** [comment]
Cool, looks good to me.

**Comment by aaime** [comment]
Well, let's merge and see what the build server thinks of it (Travis liked it)

**Comment by jonatascastro12** [comment]
Hello, guys!
Could you tell me if this commit resolved the problem of calling WMS layer getMap with a style with workspace assigned without declaring the workspace prefix (actually, the GWC work)? e.g:

Workspace: TEST
Style: TEST:point
Layer: TEST:city

`GET http://localhost/geoserver/wms?BBOX=...&TRANSPARENT=TRUE&EXCEPTIONS=application/vnd.ogc.se_xml&VERSION=1.1.1&FORMAT=image/png&SERVICE=WMS&HEIGHT=512&LAYERS=TEST:city&REQUEST=GetMap&**STYLES=point**&WIDTH=768&SRS=EPSG:900913` **DO NOT WOK**
*The GWC calls this request and it fails because workspace is not declared in styles param.*


`GET http://localhost/geoserver/wms?BBOX=...&TRANSPARENT=TRUE&EXCEPTIONS=application/vnd.ogc.se_xml&VERSION=1.1.1&FORMAT=image/png&SERVICE=WMS&HEIGHT=512&LAYERS=TEST:city&REQUEST=GetMap&**STYLES=TEST:point**&WIDTH=768&SRS=EPSG:900913` **IT WORKS**

I've been seeing the discussion about this issue, but I couldn't find if there is a transparent solution for this.

**Comment by smithkm** [comment]
This commit is what allows the use of a workspace prefix on style names in the first place.  Without it, both of those requests would fail.

**Comment by jonatascastro12** [comment]
@smithkm Thank you very much for replying! I really appreciate that!

Can you tell me if the `WORSKPACE param` works with **GWC standalone** (I can't make it work. GWC isn't translating `/geoserver/wms...&workspace=TEST` to `/geoserver/TEST/wms...`)?
(It would also answer my question here: https://github.com/geoserver/geoserver/pull/622)

**Comment by smithkm** [comment]
@jonatascastro12 Pull request comments really aren't the right place for this.  I suggest you start a thread on one of the user mailing lists.  Normally stand alone GWC questions should go to the GWC users list, but since your question is specifically about GeoServer workspaces, the GeoServer mailing list might be more appropriate.

https://lists.sourceforge.net/lists/listinfo/geoserver-users
https://lists.sourceforge.net/lists/listinfo/geowebcache-users

**Pull request merged**